include/glow/Backends/Backend.h

```diff
@@ -86,6 +85,17 @@ class Backend {
 /// Create a backend of kind \p kind.
 Backend *createBackend(BackendKind backendKind);
 
+// Backends that use Glow low-level IR should inherit from this class. It allows
+// for unit tests to create low-level IR to compile and run.
+class BackendUsingGlowIR : public Backend {
+public:
```

Comment:
I would have expected an override of the compile method for the Function here.

Reply:
As in providing the `compile(Function)` implementation for the Interpreter, OpenCL, and CPU backends? This would require a call into `generateAndOptimizeIR()`. This means Backend would depend indirectly depend on IR (via depending on the Optimizer) which we do not want.

Reply:
> As in providing the compile(Function) implementation for the Interpreter, OpenCL, and CPU backends?

Yes.

> This means Backend would depend indirectly depend on IR (via depending on the Optimizer) which we do not want.

Oh, but this is fine to have this dependency on the CPU and OpenCL backend, I see. The split seems artificial but okay. What I mean is we could have a backend lib and a backend with glow ir lib, with the dependency going like this:
```
libBackend -> libBackendWithGlowIR (shared code) -> libCPUbackend
                                        \
                                         +> libOCLbackend
```
Instead of
```
libBackend (with backendWithGlowIR) -> libCPUbackend (duplicate code)
         \
          +> libOCLbackend (duplicate code)
```

Reply:
Ah yeah, that could work. I'm not sure it's worth the added complexity though since it's a pretty small amount of shared code, and `libBackendWithGlowIR` would be tiny, at least right now. Do you think it's worth it?

Reply:
I agree, this is probably not worth the additional complexity, at least for now.
I guess if we end up with a bunch of backend using the IR, then we can rethink our strategy.